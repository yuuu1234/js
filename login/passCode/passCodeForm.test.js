import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import PassCodeForm from './passCodeForm';

describe('PassCodeForm', () => {
  const props = {
    navigation: {
      navigate: jest.fn(),
    },
    loginStore: {
      saveInputChange: jest.fn(),
    },
  };
  const component = <PassCodeForm {...props} />;
  let wrapper;

  beforeEach(() => {
    wrapper = mount(component);
  });

  it('should render component correctly', () => {
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });

  describe('handleCodeInput', () => {
    it('should call saveInputChange with joined string', () => {
      wrapper.find({ testID: 'input-pass-code' }).props().onInputChange('', [1, 2, 3]);

      expect(props.loginStore.saveInputChange).toHaveBeenCalledWith({ passCode: '123' });
    });
  });
});
