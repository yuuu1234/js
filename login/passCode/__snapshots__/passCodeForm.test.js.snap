// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PassCodeForm should render component correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "width": "70%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#003D9A",
            "fontSize": 22,
            "fontWeight": "bold",
            "marginTop": 31,
            "textAlign": "center",
            "textTransform": "capitalize",
          }
        }
      >
        Please Enter Your Passcode.
      </Text>
    </View>
  </View>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": 23.75,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "borderColor": "#97B5E2",
              "borderRadius": 50,
              "borderWidth": 1,
              "height": 13,
              "marginLeft": 13,
              "marginRight": 13,
              "width": 13,
            },
            Object {
              "backgroundColor": "#97B5E2",
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "borderColor": "#97B5E2",
              "borderRadius": 50,
              "borderWidth": 1,
              "height": 13,
              "marginLeft": 13,
              "marginRight": 13,
              "width": 13,
            },
            Object {
              "backgroundColor": "#97B5E2",
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "borderColor": "#97B5E2",
              "borderRadius": 50,
              "borderWidth": 1,
              "height": 13,
              "marginLeft": 13,
              "marginRight": 13,
              "width": 13,
            },
            Object {
              "backgroundColor": "#97B5E2",
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "borderColor": "#97B5E2",
              "borderRadius": 50,
              "borderWidth": 1,
              "height": 13,
              "marginLeft": 13,
              "marginRight": 13,
              "width": 13,
            },
            Object {
              "backgroundColor": "#97B5E2",
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "borderColor": "#97B5E2",
              "borderRadius": 50,
              "borderWidth": 1,
              "height": 13,
              "marginLeft": 13,
              "marginRight": 13,
              "width": 13,
            },
            Object {
              "backgroundColor": "#97B5E2",
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "borderColor": "#97B5E2",
              "borderRadius": 50,
              "borderWidth": 1,
              "height": 13,
              "marginLeft": 13,
              "marginRight": 13,
              "width": 13,
            },
            Object {
              "backgroundColor": "#97B5E2",
            },
          ]
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "height": 345,
          "justifyContent": "center",
          "marginTop": 23.75,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
            "marginTop": 19,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": 75,
              "justifyContent": "center",
              "marginLeft": 14.25,
              "marginRight": 14.25,
              "width": 75,
            }
          }
        >
          <View
            accessibilityLabel="login-passCode-1"
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#003D9A",
                  "borderRadius": 50,
                  "borderWidth": 1,
                  "display": "flex",
                  "height": 75,
                  "justifyContent": "center",
                  "width": 75,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#003D9A",
                    "fontSize": 36,
                    "lineHeight": 43,
                    "textAlign": "center",
                  }
                }
              >
                1
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": 75,
              "justifyContent": "center",
              "marginLeft": 14.25,
              "marginRight": 14.25,
              "width": 75,
            }
          }
        >
          <View
            accessibilityLabel="login-passCode-2"
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#003D9A",
                  "borderRadius": 50,
                  "borderWidth": 1,
                  "display": "flex",
                  "height": 75,
                  "justifyContent": "center",
                  "width": 75,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#003D9A",
                    "fontSize": 36,
                    "lineHeight": 43,
                    "textAlign": "center",
                  }
                }
              >
                2
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": 75,
              "justifyContent": "center",
              "marginLeft": 14.25,
              "marginRight": 14.25,
              "width": 75,
            }
          }
        >
          <View
            accessibilityLabel="login-passCode-3"
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#003D9A",
                  "borderRadius": 50,
                  "borderWidth": 1,
                  "display": "flex",
                  "height": 75,
                  "justifyContent": "center",
                  "width": 75,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#003D9A",
                    "fontSize": 36,
                    "lineHeight": 43,
                    "textAlign": "center",
                  }
                }
              >
                3
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
            "marginTop": 19,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": 75,
              "justifyContent": "center",
              "marginLeft": 14.25,
              "marginRight": 14.25,
              "width": 75,
            }
          }
        >
          <View
            accessibilityLabel="login-passCode-4"
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#003D9A",
                  "borderRadius": 50,
                  "borderWidth": 1,
                  "display": "flex",
                  "height": 75,
                  "justifyContent": "center",
                  "width": 75,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#003D9A",
                    "fontSize": 36,
                    "lineHeight": 43,
                    "textAlign": "center",
                  }
                }
              >
                4
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": 75,
              "justifyContent": "center",
              "marginLeft": 14.25,
              "marginRight": 14.25,
              "width": 75,
            }
          }
        >
          <View
            accessibilityLabel="login-passCode-5"
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#003D9A",
                  "borderRadius": 50,
                  "borderWidth": 1,
                  "display": "flex",
                  "height": 75,
                  "justifyContent": "center",
                  "width": 75,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#003D9A",
                    "fontSize": 36,
                    "lineHeight": 43,
                    "textAlign": "center",
                  }
                }
              >
                5
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": 75,
              "justifyContent": "center",
              "marginLeft": 14.25,
              "marginRight": 14.25,
              "width": 75,
            }
          }
        >
          <View
            accessibilityLabel="login-passCode-6"
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#003D9A",
                  "borderRadius": 50,
                  "borderWidth": 1,
                  "display": "flex",
                  "height": 75,
                  "justifyContent": "center",
                  "width": 75,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#003D9A",
                    "fontSize": 36,
                    "lineHeight": 43,
                    "textAlign": "center",
                  }
                }
              >
                6
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
            "marginTop": 19,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": 75,
              "justifyContent": "center",
              "marginLeft": 14.25,
              "marginRight": 14.25,
              "width": 75,
            }
          }
        >
          <View
            accessibilityLabel="login-passCode-7"
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#003D9A",
                  "borderRadius": 50,
                  "borderWidth": 1,
                  "display": "flex",
                  "height": 75,
                  "justifyContent": "center",
                  "width": 75,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#003D9A",
                    "fontSize": 36,
                    "lineHeight": 43,
                    "textAlign": "center",
                  }
                }
              >
                7
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": 75,
              "justifyContent": "center",
              "marginLeft": 14.25,
              "marginRight": 14.25,
              "width": 75,
            }
          }
        >
          <View
            accessibilityLabel="login-passCode-8"
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#003D9A",
                  "borderRadius": 50,
                  "borderWidth": 1,
                  "display": "flex",
                  "height": 75,
                  "justifyContent": "center",
                  "width": 75,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#003D9A",
                    "fontSize": 36,
                    "lineHeight": 43,
                    "textAlign": "center",
                  }
                }
              >
                8
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": 75,
              "justifyContent": "center",
              "marginLeft": 14.25,
              "marginRight": 14.25,
              "width": 75,
            }
          }
        >
          <View
            accessibilityLabel="login-passCode-9"
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#003D9A",
                  "borderRadius": 50,
                  "borderWidth": 1,
                  "display": "flex",
                  "height": 75,
                  "justifyContent": "center",
                  "width": 75,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#003D9A",
                    "fontSize": 36,
                    "lineHeight": 43,
                    "textAlign": "center",
                  }
                }
              >
                9
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
            "marginTop": 19,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": 75,
              "justifyContent": "center",
              "marginLeft": 14.25,
              "marginRight": 14.25,
              "width": 75,
            }
          }
        >
          <View
            style={
              Object {
                "width": 75,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": 75,
              "justifyContent": "center",
              "marginLeft": 14.25,
              "marginRight": 14.25,
              "width": 75,
            }
          }
        >
          <View
            accessibilityLabel="login-passCode-0"
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#003D9A",
                  "borderRadius": 50,
                  "borderWidth": 1,
                  "display": "flex",
                  "height": 75,
                  "justifyContent": "center",
                  "width": 75,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#003D9A",
                    "fontSize": 36,
                    "lineHeight": 43,
                    "textAlign": "center",
                  }
                }
              >
                0
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": 75,
              "justifyContent": "center",
              "marginLeft": 14.25,
              "marginRight": 14.25,
              "width": 75,
            }
          }
        >
          <View
            accessibilityLabel="login-passCode-del"
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "height": 75,
                  "justifyContent": "center",
                  "width": 75,
                }
              }
            >
              <SvgMock />
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
